### ISSUE TITLE
Possible incorrect definition for oval:org.cisecurity:tst:1251 in *:def:2332

### ISSUE THREAD

**Issue by Memnarch** [CISecurity/OVALRepo#1590]
I've found an interesting result on my system. But one question first:
Lets say we have a file_object. And It's "path" property references a variable with a varcheck. When this varcheck fails, the flag of the object is supposed to be error, right? Because in combination with sets this unravels an issue here.

The rough structure (namespace is always "oval:org.cisecurity:"):
```

tst:1251
  obj:672
    set(union)
      obj:669 (will return one file_item)
      obj:334
        var:32 fails its varcheck "at least one" because its empty
```
Interpreters (not sure about the spec) will evaluate obj:334 to error. This means that the Set will evaluate to error and it seems that the test automatically evaluates to error because its object is flagged as error. The issue is, that the file_item of obj:669 would actually eval the test to true because it will match the referenced states. The test actually tries to find as many occurencies of a specific Office 2016 file as possible. By searching the Registry for different values to eval the filepaths. But because a specific registry key is missing on the second object in the set(and thus the varcheck fails) the whole test tilts out as error.

**Comment by solind** [comment]
As you note obj:334 would have a flag of "error".  However, it's possible that an interpreter would would assign a flag of "incomplete", not "error", for the set object (obj:672 in your example).

I believe that it should be considered a best practice to construct OVAL content in such a way as to check for the existence of any object used by a variable, at a level high enough that it can logically prevent the "error" status resulting from the "does not exist" object flag from propagating to the top of the definition result.

**Comment by Memnarch** [comment]
Actually the [doc ](https://github.com/OVALProject/Language/blob/5.11.2/specifications/oval-language-specification.docx)describes a quite detailed flag-table to use based on the operator(Union, Intersect, Complement) and the flag states of the first and second object. At least from my understanding(hence the "not sure about the spec"). At least from that table, i'm not sure using sets is good in any case(when it comes to union). Or depending on the usecase "at least one".

**Comment by Memnarch** [comment]
Oh regarding the interesting result, it was more about this whole situation, and that oval makes it a bit harder than it needs to be, to capture/eval data. Sorry for being confusing at this. Because my Office 2016 installation is missing the "ClickOn/ToRun" Registrykey that is looked at below var:32, that's how i ended up here.

Maybe it's safer to simply split that test in 2, one for each object in the set and have the Tests be childs of an OR Criteria.

Regarding extending the specification, is there a process to gather ideas on if/how to extend it?

**Comment by Memnarch** [comment]
Yes, the overall result is an Error. Thanks for the hints. I'll create an issue.
